### PR TITLE
creating ecommerce management application

### PR DESCRIPTION
**Pull request by netabirami**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,13 +21,17 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-data-jpa</artifactId>
 		</dependency>
+<!--		<dependency>-->
```
Comment (f4r33) [inline pom.xml:24]: Remove comments

File: src/main/java/com/zalando/ECommerceManagement/userManagement/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@Validated
+@RequestMapping("/users")
+public class UserController {
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/controller/UserController.java:15]: Can you please format the code for this class by clicking ⌘⌥L (macOS) / Ctrl+Alt+L (Windows/Linux

File: src/main/java/com/zalando/ECommerceManagement/userManagement/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    private UserService userService;
+
+   @GetMapping
+    public List<User> getAllUsers(){
+       return userService.getAllUsers();
+   }
+
+   @PostMapping
+    public User addUser ( @Valid  @RequestBody User user){
+//       String encodedPassword = new BCryptPasswordEncoder().encode(user.getPassword());
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/controller/UserController.java:26]: Remove comments

File: src/main/java/com/zalando/ECommerceManagement/userManagement/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+   @Autowired
+    private UserService userService;
+
+   @GetMapping
+    public List<User> getAllUsers(){
+       return userService.getAllUsers();
+   }
+
+   @PostMapping
+    public User addUser ( @Valid  @RequestBody User user){
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/controller/UserController.java:25]: formatting

File: src/main/java/com/zalando/ECommerceManagement/userManagement/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.zalando.ECommerceManagement.userManagement.repository;
+
+import com.zalando.ECommerceManagement.userManagement.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User,Integer> {
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/repository/UserRepository.java:8]: add space after `<User,`

File: src/main/java/com/zalando/ECommerceManagement/userManagement/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import com.zalando.ECommerceManagement.userManagement.model.User;
+import com.zalando.ECommerceManagement.userManagement.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
+    @Autowired
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/service/UserService.java:12]: use constructor autowiring

File: src/main/java/com/zalando/ECommerceManagement/userManagement/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    private UserRepository userRepository;
+
+    public List<User> getAllUsers(){
+        return userRepository.findAll();
+    }
+
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/service/UserService.java:18]: extra space

File: src/main/java/com/zalando/ECommerceManagement/userManagement/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/service/UserService.java:23]: extra space

File: src/main/java/com/zalando/ECommerceManagement/userManagement/exception/GlobalExceptionHandler.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    public Map<String, String> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
+        Map<String, String> errors = new HashMap<>();
+        ex.getBindingResult().getAllErrors().forEach((error) -> {
+            String fieldName = ((FieldError) error).getField();
+            String errorMessage = error.getDefaultMessage();
+            errors.put(fieldName, errorMessage);
+        });
+        return errors;
+    }
+    @ExceptionHandler({InvalidMethodArgumentException.class})
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/exception/GlobalExceptionHandler.java:28]: add empty line before this

File: src/main/java/com/zalando/ECommerceManagement/userManagement/exception/InvalidMethodArgumentException.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package com.zalando.ECommerceManagement.userManagement.exception;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
```
Comment (f4r33) [inline src/main/java/com/zalando/ECommerceManagement/userManagement/exception/InvalidMethodArgumentException.java:10]: Do you need both `@AllArgsConstructor` and `@NoArgsConstructor`?

File: src/test/java/com/zalando/ECommerceManagement/userManagement/controller/UserControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        mockMvc.perform(post("/users")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonRequest))
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.jsonPath("$." + errorKey).value(errorMessage));
+
+        verify(userService, times(0)).createUser(any(User.class));
+    }
+
+    static Stream<Arguments> getUserInput() {
```
Comment (f4r33) [inline src/test/java/com/zalando/ECommerceManagement/userManagement/controller/UserControllerTest.java:94]: we can make this private

**Review by f4r33 — CHANGES_REQUESTED** [review]
Address the comments please 😄

**Comment by f4r33** [comment]
In the future when you create a PR can you create links to the ticket in Jira, please? Also, the name for the branch should be more descriptive. Ideally, we should have one branch per ticket and following the naming convention [ticket-id]-short-name. So for example, for [this ticket](https://zalando-abirami-project.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-13) the name would be SCRUM-13-user-api

File: pom.xml (hunk 1 above)
Comment (faruka-startsteps) [reply pom.xml:24]: still not removed

File: src/main/java/com/zalando/ECommerceManagement/userManagement/exception/GlobalExceptionHandler.java (hunk 9 above)
Comment (faruka-startsteps) [reply src/main/java/com/zalando/ECommerceManagement/userManagement/exception/GlobalExceptionHandler.java:28]: still no empty line

**Pull request merged**